src/librbd/internal.cc

```diff
@@ -3374,7 +3378,8 @@ namespace librbd {
   }
 
   int aio_read(ImageCtx *ictx, const vector<pair<uint64_t,uint64_t> >& image_extents,
-	       char *buf, bufferlist *pbl, AioCompletion *c, int op_flags)
+	       char *buf, bufferlist *pbl, AioCompletion *c, int op_flags,
+	       int force_no_cache)
```

Comment:
"force_no_cache" is used to bypass ObjectCacher when RBD cache is enabled. If CopyupRequest::read_from_parent accesses ObjectCacher, there will be deadlock on cache_lock.